app/config/errors.php

```diff
@@ -75,17 +78,19 @@
     ],
     Exception::GENERAL_CURSOR_NOT_FOUND => [
         'name' => Exception::GENERAL_CURSOR_NOT_FOUND,
-        'description' => 'The cursor is invalid. This can happen if the item represented by the cursor has been deleted.',
+        'description' => 'The cursor used in pagination is invalid. This can happen if the item represented by the cursor has been deleted. Retry this request using another item from the previous as the cursor.',
         'code' => 400,
     ],
     Exception::GENERAL_SERVER_ERROR => [
+        // TODO: This is used often. Can we take a look at the usage here and make sure it's not abused?
```

Comment:
This is a generic 500. We could try to create a specific exception for each of these...or it might be better to just start with some of the most common ones.